nautobot/core/views/generic.py

```diff
@@ -47,6 +47,34 @@
 from nautobot.utilities.views import GetReturnURLMixin, ObjectPermissionRequiredMixin
 
 
+def get_relationships_errors(request, obj):
+    required_relationships = Relationship.objects.get_required_for_model(obj)
+    relationships_errors = []
+    for side, relations in required_relationships.items():
+        for relation in relations:
+
+            # Skip referencing itself
+            if ContentType.objects.get_for_model(obj) == getattr(relation, f"{relation.required}_type"):
+                continue
+
+            model_class = getattr(relation, f"{side}_type").model_class()
```

Comment:
Defensive-coding-wise, need to handle the case where `model_class` is None (e.g., relationship to a plugin model for a plugin that's not installed at present).